vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/ad_targeting.cc

```diff
@@ -48,4 +48,30 @@ SegmentList AdTargeting::GetSegments() const {
   return segments;
 }
 
+SegmentList AdTargeting::GetInterestSegments() const {
+  SegmentList segments;
+
+  if (!features::IsTextClassificationEnabled()) {
+    return segments;
+  }
+
+  const ad_targeting::model::TextClassification text_classification_model;
+  segments = text_classification_model.GetSegments();
```

Comment:
We have unnecessary vector copy here. 
Could we avoid it by just `return text_classification_model.GetSegments();` ?

Reply:
I was following the pattern for copy elision, but happy to change. @tmancey wdyt?

Reply:
should actually be using optional - but based on talking with @tmancey this code is obsolete because we will be returning a "user model" data structure containing all segments.